### DISCUSSION TITLE
How may I optimize my cool stereograph card gif animation loop?

### DISCUSSION THREAD

**Discussion by owntheweb-archive** [ImageMagick/ImageMagick#4785] [Help] [answered]
Hello gurus,
I've extracted left and right views from a 1890's stereograph card ([source](https://www.loc.gov/pictures/collection/stereo/item/2001698233/)).

left:
![stereo 1s01904 vr left](https://user-images.githubusercontent.com/611860/151714839-49af073e-85f1-494c-ba70-cbcfc62847b8.png)

right:
![stereo 1s01904 vr right](https://user-images.githubusercontent.com/611860/151714831-9b72fdd7-6152-4e15-9954-29776b76e580.png)

I can morph the two together for a great preview based on [docs](https://legacy.imagemagick.org/Usage/anim_mods/#morph):

```
convert stereo.1s01904.tif.vr.left.png stereo.1s01904.tif.vr.right.png -morph 5 -duplicate 1,-2-1 -resize 450x450 -set delay '%[fx:(t>0&&t<n-1)?10:240]' output.gif
```

output (neat! Almost there...):

![stereo 1s01904 tif morph](https://user-images.githubusercontent.com/611860/151714807-b605320b-9ab0-41d7-ba2e-b50542f42579.gif)

The challenge though is that the animation looks a bit choppy. Do you know of a way I can make it animate to one side, pause, animate back, pause? It feels like this is what should be happening based on doc demos, yet I'm likely missing a step. At the moment it's animating to one side, pausing, moving one more frame (huh), pausing, then moving back and forth.

Thanks for taking a look in advance!

**Comment by snibgo** [comment, marked as answer]
As the linked documentation says, you get a pause at the beginning and end of the total animation, which contains the forwards and reverse motions. This is because you duplicate, and then set the delay.

In the linked doc, the command for "rose_flip_anim.gif" does those operations in the opposite order. So it builds the forward motion, sets the delay so both ends have a pause, and _then_ duplicates in the reverse order.

**Reply by owntheweb-archive** [reply]
That did the trick. Thank you.
![stereo 1s01904 tif morph](https://user-images.githubusercontent.com/611860/151721118-338cd7cf-7bcb-4416-9396-b8318de4e699.gif)
